internal/k8s/reconciler/utils.go

```diff
@@ -97,12 +103,18 @@ func routeAllowedForListenerNamespaces(gatewayNS string, allowedRoutes *gw.Allow
 	case gw.NamespacesFromSame:
 		return gatewayNS == route.GetNamespace(), nil
 	case gw.NamespacesFromSelector:
-		ns, err := metav1.LabelSelectorAsSelector(namespaceSelector.Selector)
+		namespaceSelector, err := metav1.LabelSelectorAsSelector(namespaceSelector.Selector)
+		if err != nil {
+			return false, fmt.Errorf("error parsing label selector: %w", err)
+		}
+
+		// retrieve the route's namespace and determine whether selector matches
+		namespace, err := c.GetNamespace(context.TODO(), types.NamespacedName{Name: route.GetNamespace()})
```

Comment:
```suggestion
		namespace, err := c.GetNamespace(ctx, types.NamespacedName{Name: route.GetNamespace()})
```
Assuming this was a relic prior to all the context plumbing.

Reply:
Thanks for catching this. I literally plumbed the context down _just to replace this_.

![](https://media.giphy.com/media/yvBAuESRTsETqNFlEl/giphy.gif)